pay_ios/ios/Classes/UtilityExtensions.swift

```diff
@@ -0,0 +1,225 @@
+      "billingContact": billingContact?.toDictionary(),
+      "shippingContact": shippingContact?.toDictionary(),
+      "shippingMethod": shippingMethod?.toDictionary(),
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PKPaymentMethod`.
+extension PKPaymentMethod {
+
+  /// Creates a `Dictionary` representation of the `PKPaymentMethod` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "displayName": displayName as Any?,
+      "network": network?.rawValue,
+      "type": type.rawValue
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PKContact`.
+extension PKContact {
+
+  /// Creates a `Dictionary` representation of the `PKContact` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "name": name?.toDictionary() as Any?,
+      "emailAddress": emailAddress,
+      "phoneNumber": phoneNumber?.stringValue,
+      "postalAddress": postalAddress?.toDictionary()
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PersonNameComponents`.
+extension PersonNameComponents {
+
+  /// Creates a `Dictionary` representation of the `PersonNameComponents` object.
+  public func toDictionary() -> [String: Any]? {
+    let dict = [
+      "namePrefix": (namePrefix != nil && !namePrefix!.isEmpty) ? namePrefix : nil as Any?,
+      "givenName": (givenName != nil && !givenName!.isEmpty) ? givenName : nil,
+      "middleName": (middleName != nil && !middleName!.isEmpty) ? middleName : nil,
+      "familyName": (familyName != nil && !familyName!.isEmpty) ? familyName : nil,
+      "nameSuffix": (nameSuffix != nil && !nameSuffix!.isEmpty) ? nameSuffix : nil,
+      "nickname": (nickname != nil && !nickname!.isEmpty) ? nickname : nil,
+      "phoneticRepresentation": phoneticRepresentation?.toDictionary(),
+    ].compactMapValues { $0 }
+
+    return !dict.isEmpty ? dict : nil
+  }
+}
+
+/// A set of utility methods associated to `CNPostalAddress`.
+extension CNPostalAddress {
+
+  /// Creates a `Dictionary` representation of the `CNPostalAddress` object.
+  public func toDictionary() -> [String: Any] {
+    return [
+      "street": !street.isEmpty ? street : nil,
+      "subLocality": !subLocality.isEmpty ? subLocality : nil,
+      "city": !city.isEmpty ? city : nil,
+      "subAdministrativeArea": !subAdministrativeArea.isEmpty ? subAdministrativeArea : nil,
+      "state": !state.isEmpty ? state : nil,
+      "postalCode": !postalCode.isEmpty ? postalCode : nil,
+      "country": !country.isEmpty ? country : nil,
+      "isoCountryCode": !isoCountryCode.isEmpty ? isoCountryCode : nil
+    ].compactMapValues { $0 }
+  }
+}
+
+/// A set of utility methods associated to `PKShippingMethod`.
+extension PKShippingMethod {
+
+  /// Creates a `Dictionary` representation of the `PKShippingMethod` object.
+  public func toDictionary() -> [String: Any?] {
+    return [
```

Comment:
Why did you decide not to trim `nil` values in this one?

Reply:
Good catch! I'll fix that